src/python/grpcio_tests/tests/unit/_dns_resolver_test.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2019 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for an actual dns resolution."""
+
+import unittest
+import logging
+import six
+
+import grpc
+from tests.unit import test_common
+
+
+class DNSResolverTest(unittest.TestCase):
+
+    def test_connect_loopback(self):
+        # NOTE(https://github.com/grpc/grpc/issues/18422)
+        # In short, Gevent + C-Ares = Segfault. The C-Ares driver is not
+        # supported by custom io manager like "gevent" or "libuv". This test is
+        # considered pass if the default DNS resolver didn't trigger segfault.
+        grpc.insecure_channel('loopback6.unittest.grpc.io')
```

Comment:
Does this test fail without the fix?

Because we never make a call on this channel, and because we never try to explicitly connect, IIC this test won't attempt start connecting or resolving.

Reply:
Thank you for pointing out. I over-simplified the unit test that trying avoid the missing `IPv6` support in Kokoro (or Docker). But I found the existence of `loopback4`, and it will trigger the segfault in the same way. So, I revert to the original unit test, and use `IPv4` to connect.